api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -229,6 +231,41 @@
     public void fireEvent(Object event, Annotation... qualifiers);
 
     /**
+     * Fire an asynchronous event and notify observers.
+     *
+     * @param event      the event object
+     * @param qualifiers the event qualifiers
+     * @throws IllegalArgumentException if the runtime type of the event object contains a type variable
+     * @throws IllegalArgumentException if two instances of the same qualifier type are given
+     * @throws IllegalArgumentException if an instance of an annotation that is not a qualifier type is given
+     * @throws IllegalArgumentException if the runtime type of the event object is assignable to the type of a container
+     *                                  lifecycle event
+     * @throws FireAsyncException       if a notified observer throws an exception, it will be caught and added
+     *                                  as a suppressed exception in a {@link FireAsyncException}.
+     *                                  This exception will be thrown at the end of the notification pipeline
+     * @since 2.0
+     */
+    public CompletionStage<Object> fireAsyncEvent(Object event, Annotation... qualifiers);
```

Comment:
We can probably be more type-safe by propagating the event type to the return type.